db/db_impl/db_impl_compaction_flush.cc

```diff
@@ -917,15 +917,33 @@ Status DBImpl::CompactRange(const CompactRangeOptions& options,
                               end_with_ts);
 }
 
-Status DBImpl::IncreaseFullHistoryTsLow(ColumnFamilyData* cfd,
+Status DBImpl::IncreaseFullHistoryTsLow(ColumnFamilyHandle* column_family,
                                         std::string ts_low) {
+  ColumnFamilyData* cfd;
+  if (column_family == nullptr) {
+    cfd = default_cf_handle_->cfd();
+  } else {
+    auto cfh = static_cast_with_check<ColumnFamilyHandleImpl>(column_family);
+    assert(cfh != nullptr);
+    cfd = cfh->cfd();
+  }
+  if (cfd->user_comparator()->timestamp_size() == 0) {
+    return Status::NotSupported(
```

Comment:
Ditto. Returning `InvalidArgument()` seems better.

Reply:
Also, please check the equality between `ts_low.size()` and `cfd->user_comparator()->timestamp_size()`. Return invalid argument if unequal.
If `ts_low`'s size is shorter, we still want to capture the error here, instead of letting a subsequent test in `IncreaseFullHistoryTsLowImpl()` to detect.

Reply:
added